### DISCUSSION TITLE
SDR Direct Sampling mode

### DISCUSSION THREAD

**Discussion by habub** [merbanan/rtl_433#2371] [General]
Hi,
I have an SDR device which supports direct sampling mode (aka Q-branch).
Can I activate this mode through RTL_433?

![image](https://user-images.githubusercontent.com/8079997/217308201-5be43413-f7a5-4e7b-9c05-85d685c4e09d.png)

**Comment by zuckschwerdt** [comment]
Yes, the `-t` option supports the usual RTL-SDR options:
`direct_samp[=1]`, `offset_tune[=1]`, `digital_agc[=1]`, `biastee[=1]`

```
  [-t <settings>] apply a list of keyword=value settings for SoapySDR devices
       e.g. -t "antenna=A,bandwidth=4.5M,rfnotch_ctrl=false"
```
We should add that to the help ;)

**Reply by habub** [reply]
Thanks.
So I just add -t "direct_samp=1" in the command line?
Where can I find the available -t options? Since they're not in the help...

**Reply by zuckschwerdt** [reply]
They are now ;)
```
  [-t <settings>] apply a list of keyword=value settings to the SDR device
       e.g. for SoapySDR -t "antenna=A,bandwidth=4.5M,rfnotch_ctrl=false"
       for RTL-SDR use "direct_samp[=1]", "offset_tune[=1]", "digital_agc[=1]", "biastee[=1]"
```

**Reply by habub** [reply]
Thanks for the quick turnaround.
But were can I get a full list of supported settings and their values? I couldn't find any reference in soapy or other repo.
Also, when using "direct_samp" I didn't get feedback if it worked. I used any value (=1, 2, 3, etc.) they were all accepted.

**Reply by zuckschwerdt** [reply]
It's just the RTL-SDR options. Booleans except `direct_samp`, which is 0: Off, 1: I-ADC, 2: Q-ADC.
https://github.com/pothosware/SoapyRTLSDR/blob/master/Settings.cpp#L562
https://github.com/osmocom/rtl-sdr/blob/master/include/rtl-sdr.h#L295
